ee/packages/license/src/AirGappedRestriction.ts

```diff
@@ -0,0 +1,73 @@
+import EventEmitter from 'events';
+
+import { License } from '.';
+import { decryptStatsToken } from './token';
+
+const DAY_IN_MS = 24 * 60 * 60 * 1000;
+const NO_ACTION_PERIOD_IN_DAYS = 3;
+const WARNING_PERIOD_IN_DAYS = 7;
+
+class AirGappedRestrictionClass extends EventEmitter {
+	#restricted = false;
```

Comment:
should the initial state be `restricted = true` ? I think for the purposes we want to achieve it should be restricted by default.

we might also need a hook when adding a license so we can remove the restriction right away

Reply:
Really nice catch. This was not described in the Project whatsoever. I already asked for definitions on this. There's the possibility that new workspaces should start in the grace period, but that will require some changes, and most likely a discussion. An update to the ADR will also be necessary.

Reply:
After some discussion, this will stay like this, workspaces should be restricted by default.